chapters/functions.tex

```diff
@@ -362,7 +362,8 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
 impure function receiveRealSignal // impure function
   input HardwareDriverID id;
   output Real y;
-  external "C" y = receiveSignal(id);
+  external "C"
+    y = receiveSignal(id);
```

Comment:
Hmm... This style looks odd to me, I will have to think more about it.

Reply:
Take your time.

Reply:
Hmm... Looking at calling functions in other languages, i.e. foreign function interface https://en.wikipedia.org/wiki/Foreign_function_interface

Calling Java from SQL: EXTERNAL NAME 'some.random.name'
https://docs.oracle.com/javadb/10.10.1.2/ref/rrefsqljexternalname.html

C++ calling C: https://en.cppreference.com/w/cpp/language/language_linkage
```
extern "C" {
    int open(const char *pathname, int flags); // C function declaration
}
extern "C" int x; 
```

Haskell calling C: `foreign import ccall "set_icval = ccall(:getenv, Cstring, (Cstring,), var)anon" set_icanon :: CInt -> IO ()`
Lisp calling C: `(defcfun ("abs" c-abs) :int (n :int))`
Ada calling Fortran: pragma Import(Fortran, Exp);
Julia directly calling C: `val = ccall(:getenv, Cstring, (Cstring,), var)`
Rust calling C: 
```
extern "C" {
    fn abs(input: i32) -> i32;
}
```

**Conclusion:**

The thing that looks closest to the proposal is one of the variants for C++ calling C. However, thinking more I think it would make more sense to have it as follows (basically having external instead of algorithm):
```
impure function receiveRealSignal // impure function
  input HardwareDriverID id;
  output Real y;
external "C"
  y = receiveSignal(id);
```
However,

- I think we should have a non-normative note in the specification saying that the other formatting is also ok (because we don't want everyone to change their libraries).
- Having `external "C";` in this style was also confusing at first, but now I think that is also ok.

Reply:
Looks like good style to me, but I'll wait with making changes until after the phone meeting in case more comments would come up.

What I don't understand is why this should be discussed at all in the specification text, even non-normatively.

Reply:
Code formatting standard in Modelica Standard Library.

Reply:
The indentation has now been updated.  I recommend taking a final look at the indentation of the annotation of the external clause before resolving this conversation.